### PR TITLE
[Improve] Implement GetTopicAutoCreation

### PR DESCRIPTION
**Pull request by jiangpengcheng**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

The `GetTopicAutoCreation` endpoint is missed, needs to add it

### Modifications

Implement the `GetTopicAutoCreation` method

### Verifying this change

- [x] Make sure that the change passes the CI checks.

- [x] Added some tests for topic auto-creation

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)

### Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? (not applicable / docs / GoDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: pulsaradmin/pkg/utils/topic_auto_creation_config.go [hunk 1]
```diff
@@ -20,5 +20,5 @@ package utils
 type TopicAutoCreationConfig struct {
 	Allow      bool      `json:"allowAutoTopicCreation"`
 	Type       TopicType `json:"topicType"`
-	Partitions int       `json:"defaultNumPartitions"`
+	Partitions *int      `json:"defaultNumPartitions"`
```
Comment (jiangpengcheng) [inline pulsaradmin/pkg/utils/topic_auto_creation_config.go:23]: it's not allowed to pass the `partitions` argument when the type is `non-partitioned`, so need to set this as a pointer

**Review by RobertIndie — COMMENTED** [review]
Is it possible to add a test for it?

**Comment by jiangpengcheng** [comment]
@RobertIndie There are no tests for the `namespaces`(and other modules) at all, we may need a PIP to setup the tests structure first

**Comment by RobertIndie** [comment]
> @RobertIndie There are no tests for the `namespaces`(and other modules) at all, we may need a PIP to setup the tests structure first

I don't think we need a PIP for that. We can utilize the existing testing framework. Seems that you can just create a test go file and write the test. I have created a PR to show how to do it: https://github.com/apache/pulsar-client-go/pull/1152

Then you can refer to these scripts to run the tests: https://github.com/apache/pulsar-client-go/tree/master/scripts . It will run in the CI: https://github.com/apache/pulsar-client-go/blob/master/.github/workflows/ci.yml#L50

**Comment by jiangpengcheng** [comment]
I see, there will be a pulsar instance for testing, added some tests for topic auto-creation

**Review by nlu90 — APPROVED** [review]

**Review by RobertIndie — APPROVED** [review]

**Pull request merged**